### ISSUE TITLE
Bug: `Boxes.merge` concatenates unbatched `(N, 4, 2)` boxes along the vertex axis

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4168]
## Summary

`Boxes.merge` concatenates along `dim=1` unconditionally (`kornia/geometry/boxes.py:329` at `35e4938d`). For batched `(B, N, 4, 2)` data that is the box axis. For unbatched `(N, 4, 2)` data it is the **vertex** axis, so two unbatched objects merge into an `(N, 8, 2)` tensor and every later export silently reads the eight vertices as one box.

## Reproduction

```python
import torch
from kornia.geometry.boxes import Boxes

a = Boxes.from_tensor(torch.tensor([[1.0, 2.0, 5.0, 4.0]]), mode="xyxy")
b = Boxes.from_tensor(torch.tensor([[6.0, 3.0, 9.0, 8.0]]), mode="xyxy")
m = a.merge(b)
print(m.data.shape)          # torch.Size([1, 8, 2])  -- expected torch.Size([2, 4, 2])
print(m.to_tensor("xyxy"))   # tensor([[1., 2., 9., 8.]])  -- the union bounding box of both inputs, one row
```

Neither the constructor nor `to_tensor` validates `data.shape[-2] == 4` after the merge, so nothing raises.

## Expected

`(N, 4, 2)` + `(M, 4, 2)` → `(N + M, 4, 2)`, matching the documented batched behavior. The fix is `dim=-3` (or `dim=1 if self._is_batched else 0`). `Boxes3D` has no `merge`, so the 3D class is unaffected.

## Context

Found while reviewing #4060 (batch 4a), whose `merge` docstring and pin describe the batched case only. Not in scope there: that PR is documentation and tests only. Part of the #4020 bbox/boxes audit; disposition: separate fix PR, it does not overlap #3934/#4061.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).

**Comment by alinaqvi1129** [comment]
id like to work on this @ducha-aiki can u assign this to me

**Comment by ducha-aiki** [comment]
@alinaqvi1129 yes, take it. GitHub will not let me put a non-collaborator in the assignee field, so this comment is the claim marker.

Confirmed still live on `main` @ `ee6b713f` (torch 2.9.1):

```
unbatched merge -> shape (1, 8, 2), to_tensor("xyxy") = [[1.0, 2.0, 9.0, 8.0]]   # one box, the union
batched   merge -> shape (1, 2, 4, 2)                                            # correct
```

Three notes before you start:

**1. `dim=-3` is the fix, and it is free for the batched path.** For 4-D `(B, N, 4, 2)` data `dim=-3` *is* `dim=1`, so batched behavior is unchanged by construction. Prefer it over `dim=1 if self._is_batched else 0`.

**2. There is one internal caller and it is safe — but show that.** `kornia/augmentation/_2d/mix/mosaic.py:146` calls `merge(..., inplace=True)`. It operates on batched boxes (it indexes `_box._data` on dim 0 with `to_apply` and the permutation), so `dim=-3` leaves it byte-identical. Please still run `pixi run test-module tests/augmentation -k mosaic` and paste the count in the PR body — that is the evidence, not the argument.

**3. The pin must assert values, not only the shape.** `(1, 8, 2)` → `(2, 4, 2)` is a shape change, so a shape assertion does fail on base, but a shape-only test would still pass against a wrong reordering. Assert the full `to_tensor("xyxy")` result for the unbatched merge — with the numbers above, base returns the single union row `[[1, 2, 9, 8]]` and the fix must return both rows — and keep a batched case alongside it so a future change cannot flip the axis back.

Also update the `merge` docstring at `kornia/geometry/boxes.py:329`: it only describes the `(B, N, 4, 2)` case, which is how this stayed invisible. `Boxes3D` has no `merge`, so there is nothing to mirror on the 3D side.

Standard asks: run the new test with `kornia/` swapped to the merge-base and paste the fail-on-base output, and add a `## Unreleased` → `Bug fixes` entry in `CHANGELOG.md` citing your PR number.

Posted on behalf of @ducha-aiki by Claude (Opus 5).
